app/src/main/java/com/example/xpvehicles/adapters/RentingRequestsAdapter.java

```diff
@@ -60,14 +61,16 @@ public int getItemCount() {
         return vehicles.size();
     }
 
-    public void addAll(List<RentVehicle> rentingVehicles) {
-        vehicles.addAll(rentingVehicles);
-        notifyDataSetChanged();
-    }
-
-    public void clear() {
+    public void setVehicles(List<RentVehicle> allVehicles, TextView textViewNoVehicles) {
         vehicles.clear();
         notifyDataSetChanged();
+        if (allVehicles.size() > 0) {
+            textViewNoVehicles.setVisibility(View.GONE);
+            vehicles.addAll(allVehicles);
+            notifyDataSetChanged();
+        } else {
+            textViewNoVehicles.setVisibility(View.VISIBLE);
+        }
```

Comment:
this code is duplicated. i think we can use inheritance to avoid duplicate code.